src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtTokenProvider.java

```diff
@@ -14,30 +15,38 @@
 import uk.gov.hmcts.reform.sandl.snlapi.security.model.UserPrincipal;
 
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 
 @Component
 public class JwtTokenProvider {
 
     private static final Logger logger = LoggerFactory.getLogger(JwtTokenProvider.class);
+    public static final String MAX_EXPIRY_DATE = "maxExpiryDate";
 
     @Value("${management.security.jwtSecret}")
     private String jwtSecret;
 
     @Value("${management.security.jwtExpirationInMs}")
     private int jwtExpirationInMs;
 
-    public String generateToken(Authentication authentication) {
+    @Value("${management.security.jwtMaxExpirationInMs}")
+    private int jwtMaxExpirationInMs;
 
+    public String generateToken(Authentication authentication, Date maxExpiryDate) {
```

Comment:
what is the difference between this function and S2SAuthenTicationService.TokenCreator.createToken - do not both functions do the same? DRY

Reply:
S2S (service to service) creates a different kind of token, for s2s actions.
Can it be reused, as a kind of a base class? yes.
Why it wasn't? because I haven't even looked into it and treated as a separate element of the system.